src/Symfony/Component/Security/Core/User/LdapUserProvider.php

```diff
@@ -113,30 +118,31 @@ public function supportsClass($class)
      */
     protected function loadUser($username, Entry $entry)
     {
-        $password = $this->getPassword($entry);
+        $password = $this->getAttributeValue($entry, $this->passwordAttribute);
 
         return new User($username, $password, $this->defaultRoles);
     }
 
     /**
-     * Fetches the password from an LDAP entry.
+     * Fetches a required unique attribute value from an LDAP entry.
      *
      * @param null|Entry $entry
+     * @param string     $attribute
      */
-    private function getPassword(Entry $entry)
+    private function getAttributeValue(Entry $entry, $attribute)
     {
-        if (null === $this->passwordAttribute) {
+        if (null === $attribute) {
```

Comment:
@quentinus95 Actually, that's already the case. There are cases where we expect to not have any password attribute (`$passwordAttribute === null`). In this case, the password is set to null.

This actually happens in most LDAP configurations, where the password is not returned by the LDAP server. In this case, it is actually expected to authenticate the user using the LDAP bind mechanism, which is handled in the `LdapBindAuthenticator`, which is used by `form_login_ldap` and `http_basic_ldap`.

What I mean here is that we only need to move the null checking part inside `loadUser()` as the only case where we need this check is for the password handling.